src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs

```diff
@@ -161,18 +160,18 @@ private static object Locale_Parser(string name, PatternTable table)
 
             if (name.Length == 2)
             {
-                return name;
+                return name.ToString();
```

Comment:
Same concern as above, when does this run? If it runs always, then this will defeat the purpose of the original change.

Reply:
This code path will be executed only when parsing resource assemblies because they have `{locale}` path segment in them. https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs#L486-L497

Reply:
Kartheek is correct. 
This only runs when we actually end up using the string. 
The challenge is deduplication, which is handled here already. This change is safe.